packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -54,6 +54,11 @@ import 'image_provider.dart';
 /// ```
 /// {@end-tool}
 ///
+/// Note that the [shape] or the [borderRadius] won't clip the children of the
+/// decorated [Container]. If the clip is required, insert a clip widget
+/// (e.g., [ClipRect], [ClipRRect], [ClipPath]) as the child of the [Container].
```

Comment:
We should probably repeat this explanation on the doc for the `shape` and `borderRadius` property below (line 171 and 138).

Reply:
Done.